fuzz.go

```diff
@@ -0,0 +1,29 @@
+// +build gofuzz
+
+package sftp
+
+import (
+	"bytes"
+	"io/ioutil"
+)
+
+type sink struct{}
+
+func (*sink) Close() error                { return nil }
+func (*sink) Write(p []byte) (int, error) { return len(p), nil }
+
+var devnull = &sink{}
+
+// To run: go-fuzz-build && go-fuzz
+func Fuzz(data []byte) int {
+	r := ioutil.NopCloser(bytes.NewReader(data))
+	c, err := NewClientPipe(r, devnull)
+	if err == nil {
+		c.Close()
```

Comment:
Yes. go-fuzz is not the most user-friendly tool, but it's quite powerful. There's a [draft proposal](https://go.googlesource.com/proposal/+/master/design/draft-fuzzing.md) for fuzz testing in the stdlib, but for now, go-fuzz is the best we've got.

Re: only testing recvVersion, @davecheney didn't want to test a private API. This does exercise the main piece of code that the client and server share, but it doesn't get to see all the inlined packet unmarshaling in the client. Getting the fuzzer to test that has proven very difficult, since you need to simulate an entire conversation.

Reply:
:+1: sounds good